torch/_inductor/lowering.py

```diff
@@ -1381,11 +1386,12 @@ def inner_fn(idx):
         if zero_points.dtype != torch.float32:
             zero_point = ops.to_dtype(zero_point, torch.float32)
         val = ops.sub(ops.to_dtype(input, torch.float32), zero_point) * scale
+        val = ops.to_dtype(val, out_dtype)
```

Comment:
Checking the generated code, we won't see the redundant cast if `out_dtype` is already `torch.float32`.